src/mongocxx/test/spec/transactions.cpp

```diff
@@ -378,7 +378,7 @@ void run_transactions_tests_in_file(const std::string& test_path) {
         };
 
         if (test["expectations"]) {
-            apm_checker.compare(test["expectations"].get_array().value, false, visitor);
+            apm_checker.compare(test["expectations"].get_array().value, true, visitor);
```

Comment:
`has_collection` calls `list_collection` under the hood. This gives us two extra `command_started_event`s: one for `assertCollectionExists` and one for `assertCollectionNotExists`. Not sure if there's a to avoid this though. Open to suggestions.

Reply:
Can we run `assertCollectionExists`, `assertCollectionNotExists`, `assertIndexExists`, and `assertIndexNotExists` on a separate mongocxx::client? There is some precedent for that. `targetedFailPoint` is meant to be run from a different client for that very reason:

> Tests that use the "targetedFailPoint" operation do not include
``configureFailPoint`` commands in their command expectations. Drivers MUST
ensure that ``configureFailPoint`` commands do not appear in the list of logged
commands, either by manually filtering it from the list of observed commands or
by using a different MongoClient to execute ``configureFailPoint``.

Reply:
Good call. That worked!

Reply:
Nice!